packages/flutter/lib/src/material/i18n/date_localizations.dart

```diff
@@ -0,0 +1,2936 @@
+      r'''12'''
+    ],
+    'MONTHS': const <dynamic>[
+      r'''一月''',
+      r'''二月''',
+      r'''三月''',
+      r'''四月''',
+      r'''五月''',
+      r'''六月''',
+      r'''七月''',
+      r'''八月''',
+      r'''九月''',
+      r'''十月''',
+      r'''十一月''',
+      r'''十二月'''
+    ],
+    'STANDALONEMONTHS': const <dynamic>[
+      r'''一月''',
+      r'''二月''',
+      r'''三月''',
+      r'''四月''',
+      r'''五月''',
+      r'''六月''',
+      r'''七月''',
+      r'''八月''',
+      r'''九月''',
+      r'''十月''',
+      r'''十一月''',
+      r'''十二月'''
+    ],
+    'SHORTMONTHS': const <dynamic>[
+      r'''1月''',
+      r'''2月''',
+      r'''3月''',
+      r'''4月''',
+      r'''5月''',
+      r'''6月''',
+      r'''7月''',
+      r'''8月''',
+      r'''9月''',
+      r'''10月''',
+      r'''11月''',
+      r'''12月'''
+    ],
+    'STANDALONESHORTMONTHS': const <dynamic>[
+      r'''1月''',
+      r'''2月''',
+      r'''3月''',
+      r'''4月''',
+      r'''5月''',
+      r'''6月''',
+      r'''7月''',
+      r'''8月''',
+      r'''9月''',
+      r'''10月''',
+      r'''11月''',
+      r'''12月'''
+    ],
+    'WEEKDAYS': const <dynamic>[
+      r'''星期日''',
+      r'''星期一''',
+      r'''星期二''',
+      r'''星期三''',
+      r'''星期四''',
+      r'''星期五''',
+      r'''星期六'''
+    ],
+    'STANDALONEWEEKDAYS': const <dynamic>[
+      r'''星期日''',
+      r'''星期一''',
+      r'''星期二''',
+      r'''星期三''',
+      r'''星期四''',
+      r'''星期五''',
+      r'''星期六'''
+    ],
+    'SHORTWEEKDAYS': const <dynamic>[
+      r'''周日''',
+      r'''周一''',
+      r'''周二''',
+      r'''周三''',
+      r'''周四''',
+      r'''周五''',
+      r'''周六'''
+    ],
+    'STANDALONESHORTWEEKDAYS': const <dynamic>[
+      r'''周日''',
+      r'''周一''',
+      r'''周二''',
+      r'''周三''',
+      r'''周四''',
+      r'''周五''',
+      r'''周六'''
+    ],
+    'NARROWWEEKDAYS': const <dynamic>[
+      r'''日''',
+      r'''一''',
+      r'''二''',
+      r'''三''',
+      r'''四''',
+      r'''五''',
+      r'''六'''
+    ],
+    'STANDALONENARROWWEEKDAYS': const <dynamic>[
+      r'''日''',
+      r'''一''',
+      r'''二''',
+      r'''三''',
+      r'''四''',
+      r'''五''',
+      r'''六'''
+    ],
+    'SHORTQUARTERS': const <dynamic>[
+      r'''1季度''',
+      r'''2季度''',
+      r'''3季度''',
+      r'''4季度'''
+    ],
+    'QUARTERS': const <dynamic>[
+      r'''第一季度''',
+      r'''第二季度''',
+      r'''第三季度''',
+      r'''第四季度'''
+    ],
+    'AMPMS': const <dynamic>[r'''上午''', r'''下午'''],
+    'DATEFORMATS': const <dynamic>[
+      r'''y年M月d日EEEE''',
+      r'''y年M月d日''',
+      r'''y年M月d日''',
+      r'''y/M/d'''
+    ],
+    'TIMEFORMATS': const <dynamic>[
+      r'''zzzz ah:mm:ss''',
+      r'''z ah:mm:ss''',
+      r'''ah:mm:ss''',
+      r'''ah:mm'''
+    ],
+    'AVAILABLEFORMATS': null,
+    'FIRSTDAYOFWEEK': 6,
+    'WEEKENDRANGE': const <dynamic>[5, 6],
+    'FIRSTWEEKCUTOFFDAY': 5,
+    'DATETIMEFORMATS': const <dynamic>[
+      r'''{1} {0}''',
+      r'''{1} {0}''',
+      r'''{1} {0}''',
+      r'''{1} {0}'''
+    ],
+  },
+};
```

Comment:
Why maps instead of just hard-coded classes? It would probably be a lot more efficient that way.

Reply:
We are constructing the [DateSymbols class](https://www.dartdocs.org/documentation/intl/0.15.2/date_symbols/DateSymbols-class.html) from this data once during initialization. It's objects from then on.

We could reduce initialization time if we could create `const` objects. Unfortunately, `DateSymbols` is not `const`, so it has to be initialized at runtime. I can look into that next, but it doesn't look doable as part of this PR.